### PR TITLE
Do not encode JSON objects again

### PR DESCRIPTION
**Pull request by XVincentX**
This PR makes sure to check out what kind of parameter we're treating when creating a message body.
If it's an object, then we can serialise it as JSON, but if it's not, we should just forward it to the `Asset` object. If not, we will double encode it and a `JSON.parse` conversion will not give the desired result.

### CODE REVIEW

**Comment by XVincentX** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;X9c4SgAK&#x2F;98-fix-json-examples"> Fix JSON examples</a>

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+; Install the "EditorConfig" plugin into your editor to use
+
+root = true
+
+[*]
+charset = utf-8
+insert_final_newline = true
+indent_style = space
+indent_size = 2
+trim_trailing_whitespace = true
```
Comment (smizell) [inline .editorconfig]: A [Rider](https://en.wikipedia.org/wiki/Rider_(legislation)) :)

Looks good to me though.

File: src/adapter.js [hunk 2]
```diff
@@ -695,7 +695,13 @@ export function parse({minim, source, generateSourceMap}, done) {
 
             // Responses can have bodies
             if (responseBody !== undefined) {
-              const bodyAsset = new Asset(JSON.stringify(responseBody, null, 2));
+              let formattedResponseBody = responseBody;
+
+              if (typeof(responseBody) === 'object') {
```
Comment (kylef) [inline src/adapter.js]: Would array's of objects also need to be stringified too?

Comment (smizell) [inline src/adapter.js]: Does this work? Can't valid JSON be any primitive value, from string, to array, to object? I will need to look at the Swagger spec, but I'm thinking we'll need to stringify anything other than a string (assuming a string is either a primitive string or JSON-stringified).

File: test/fixtures/refract/payload-as-string.json [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  "meta": {},
+  "attributes": {},
+  "content": [
+    {
+      "element": "category",
+      "meta": {
+        "classes": [
+          "api"
+        ],
+        "title": "Not formated example"
```
Comment (smizell) [inline test/fixtures/refract/payload-as-string.json]: Nit: formatted misspelled

File: test/fixtures/refract/x-summary-and-description.json [hunk 4]
```diff
@@ -52,7 +52,9 @@
     {
       "element": "category",
       "meta": {
-        "classes": ["api"]
+        "classes": [
+          "api"
+        ]
```
Comment (smizell) [inline test/fixtures/refract/x-summary-and-description.json]: Specific reason for this?

Comment (pksunkara) [inline test/fixtures/refract/x-summary-and-description.json]: It's how the generated JSONs are serialized.

Comment (smizell) [inline test/fixtures/refract/x-summary-and-description.json]: True, but I'm not sure why this is a change to this PR as it seems unrelated. Is it making things consistent? Or was this making tests fail?

Comment (XVincentX) [inline test/fixtures/refract/x-summary-and-description.json]: Well it was regenerated by `npm` test command. Will fix!

File: test/fixtures/refract/payload-as-string.json (hunk 3 above)
Comment (XVincentX) [inline test/fixtures/refract/payload-as-string.json]: Thanks, will fix

File: src/adapter.js (hunk 2 above)
Comment (XVincentX) [inline src/adapter.js]: Right, I will revert the condition then-

File: test/fixtures/swagger/payload-as-string.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                "results": [
+                  {
+                    "id": 12312323,
+                    "type": "article"
+                  }
+                ]
+              }
+
+        default:
+          description: Error
```
Comment (XVincentX) [inline test/fixtures/swagger/payload-as-string.yaml]: True that, will fix.

File: test/fixtures/refract/x-summary-and-description.json (hunk 4 above)
Comment (pksunkara) [inline test/fixtures/refract/x-summary-and-description.json]: I don't think this needs to be fixed.

Comment (smizell) [inline test/fixtures/refract/x-summary-and-description.json]: No problem to fix if it was auto generated. Was just curious why it was there.

Comment (smizell) [inline test/fixtures/refract/x-summary-and-description.json]: I mean, you don't need to fix it.

**Comment by XVincentX** [comment]
@smizell Thanks for the review. I've addressed all the comments; changed the condition, added new lines where needed and stripped down the fixture to the minimal possible one. Please have a new look, thanks!
